app/metal-controller-manager/internal/server/server.go

```diff
@@ -258,7 +273,94 @@ func (s *server) Heartbeat(ctx context.Context, in *api.HeartbeatRequest) (*api.
 	return resp, nil
 }
 
-func Serve(c controllerclient.Client, recorder record.EventRecorder, scheme *runtime.Scheme, autoAccept, insecureWipe bool, rebootTimeout time.Duration) error {
+func (s *server) UpdateBMCInfo(ctx context.Context, in *api.UpdateBMCInfoRequest) (*api.UpdateBMCInfoResponse, error) {
+	if in.GetBmcInfo() != nil {
+		// Create or update creds secret
+		credsSecret := &corev1.Secret{}
+		exists := true
+
+		// Fetch corresponding server
+		obj := &metalv1alpha1.Server{}
+
+		if err := s.c.Get(ctx, types.NamespacedName{Name: in.GetUuid()}, obj); err != nil {
+			return nil, err
+		}
+
+		// For auto-created BMC info, we will *always* drop the creds in default namespace
+		// This ensures they'll come along for the ride in a "default" clusterctl move based on our cluster templates.
+		if err := s.c.Get(ctx, types.NamespacedName{Namespace: corev1.NamespaceDefault, Name: in.GetUuid() + "-bmc"}, credsSecret); err != nil {
```

Comment:
nit: `in.GetUuid() + "-bmc"` might benefit from being extracted as a variable, as we use it many times in this function